### PR TITLE
fix: allow users to control tabindex on interactive components

### PR DESCRIPTION
**Pull request by jcfranco**
**Related Issue:** #4970 

## Summary

Updates the interactive component pattern to avoid the need to control `tabindex` on the host to prevent tabbing into a component's subtree.

This works by introducing a new functional component (`InteractiveContainer`) that leverages `inert` (prevent tabbing into its contents) and `display: contents` (allows us to use it as a root container without affecting layout), which should be the root container under the host. 

Updating components to this pattern should be straightforward as the styling of the helper component is bundled within the `disabled` mixin that interactive components already include and `updateHostInteraction` no longer needs its 2nd parameter. For cases that were setting the 2nd parameter (`hostIsTabbable`) all they have to do to migrate depends on the value:

1. `false` (default) – no further changes necessary
2. `true` – set `tabIndex={0}` on the host
3. predicate function – set `tabIndex={myPredicateFunction() ? 0 : -1 }` on the host
4. `"managed"` – no further changes necessary as owner/parent sets `tabIndex` on the component already

Any use of `Fragment` can be replaced by `InteractiveContainer` directly.

2 and 3 might also need to determine the tab index based on `disabled`.

### CODE REVIEW

File: packages/calcite-design-tokens/support/token-transformer/styleDictionary/formatter/utils/formatExtraOutput.ts [hunk 1]
```diff
@@ -49,7 +49,8 @@ export function formatExtraOutput(
                 const m = ensureIfArray(outputObject[`${output}.${args.platform}`]);
                 const cssProps = m.map((o) => `${o}`.replaceAll("$", "--"));
                 return Array.isArray(m)
-                  ? `${output === "light" ? `:root {${cssProps.join("")}}` : ""
+                  ? `${
```
Comment (driskull) [inline packages/calcite-design-tokens/support/token-transformer/styleDictionary/formatter/utils/formatExtraOutput.ts:52]: This just a formatting change?

**Review by driskull — APPROVED** [review]
This looks good @jcfranco ! 🎉 

We'll just have to do some testing to make sure it won't mess up any existing styles with the added container.

File: packages/calcite-design-tokens/support/token-transformer/styleDictionary/formatter/utils/formatExtraOutput.ts (hunk 1 above)
Comment (jcfranco) [reply packages/calcite-design-tokens/support/token-transformer/styleDictionary/formatter/utils/formatExtraOutput.ts:52]: Yes, I think it got formatted when the latest from `main` got merged.

**Comment by jcfranco** [comment]
> We'll just have to do some testing to make sure it won't mess up any existing styles with the added container.

Based on the description of [`display: contents`](https://developer.mozilla.org/en-US/docs/Web/CSS/display#contents), I don't foresee any issues, but worth keeping an eye out anyways. 👁️👄👁️

**Pull request merged**